application/helpers/filebin_helper.php

```diff
@@ -1,5 +1,30 @@
 <?php
 
+function expiration_duration($duration)
+{
+	$total = $duration * 60 * 60 * 24;
+	$days = floor($total / 86400);
+	$total -= $days * 86400;
+	$hours = floor($total / 3600);
+	$total -= $hours * 3600;
+	$minutes = floor($total / 60);
+	$times = array($days, $hours, $minutes);
+	$suffixes = array(' day', ' hour', ' minute');
+	$expiration = "";
+
+	for ($i = 0; $i < 3; $i++) {
+		if ($times[$i] != 0) {
+			$expiration .= $times[$i].$suffixes[$i];
```

Comment:
Better use an array here and join the elements later with ``join(", ", $elements)``. Then the rest of the code dealing with ``$expiration`` also becomes simpler.